### PR TITLE
container/remediate.py: cleanup of repository data inside remediated image

### PR DESCRIPTION
**Pull request by matusmarhefka**
* This commit adds cleanup of repository data to generated Dockerfile from
  which the hardened image is built. This cleanup should reduce the size
  of built hardened images by tens of megabytes.

### CODE REVIEW

**Comment by matusmarhefka** [comment]
Commit 442de15 fixes PEP8 issue with except.

**Comment by jan-cerny** [comment]
@matusmarhefka I think there can be a better way to do this. Starting a new container looks like a unnecessary overhead. Have you considered some other options?

**Comment by matusmarhefka** [comment]
@jan-cerny I considered only looking at the image labels, but that is not very reliable. Do you have some other suggestions?

**Comment by jan-cerny** [comment]
Hmmm, when I run yum on Fedora, it calls dnf

**Comment by matusmarhefka** [comment]
@jan-cerny That is true, it is provided by `dnf-yum` package and it links `yum` command to `dnf` command. We can hardcode container/remediate.py to use `yum`, but in the future you might need to solve this issue anyway if we want to harden also other images than RHEL/Fedora.

**Comment by jan-cerny** [comment]
@matusmarhefka That's also true. So we might want to solve this issue completely. 

There are two things that can be useful:

1. The `fix.sh` script usually installs some packages. So it often contains "apt-get install" or "dnf install". That can be easier to analyse, but not completely reliable.

2. Better option could be the fact that this script will get a path to results directory as an argument. The directory contains ARF. The ARF contains information about target container. You can parse the ARF to get the information. Most likely CPE will help you.

**Comment by matusmarhefka** [comment]
We discussed with @jan-cerny and the best solution seems to be to look at platform CPE in ARF results. Checking `fix.sh` script is unreliable as it contains code which is deciding between `dnf` or `yum` commands directly in a container during remediation.

I will update the PR to determine package manager by looking at platform CPEs in ARF results file.

**Comment by matusmarhefka** [comment]
Commit c0814e0 updates code to work with platform CPE which is obtained from ARF results file.

File: container/remediate.py [hunk 1]
```diff
@@ -27,6 +28,8 @@
 import tempfile
 import json
 import requests
+import re
+from lxml import etree
```
Comment (jan-cerny) [inline container/remediate.py]: lxml is extra library, we use standard elementree https://docs.python.org/2/library/xml.etree.elementtree.html

Comment (matusmarhefka) [reply container/remediate.py]: I was afraid you say this  :grin:

**Comment by matusmarhefka** [comment]
Commit e0d5012 uses `xml.etree.ElementTree` instead of `lxml`.

**Comment by jan-cerny** [comment]
@matusmarhefka Great! Thank you :D

**Pull request merged**